### PR TITLE
docs: various improvements across the docs

### PR DESCRIPTION
**Pull request by mistercrunch**
Review and edits / improvements across the documentation.

## Screenshot from https://pypi.org/project/apache-superset/
Screenshot 2024-05-01 at 12 24 44 PM

Bigger items:
- changed `README.md` to use fully-qualified links to images, so that it can be reused as the Introduction page at  `https://superset.apache.org/docs/`
- this should also fix missing image links on [pypi.org](https://pypi.org/project/apache-superset/)
- doing a simple `cp README.md docs/docs/` on docs `yarn build` and `yarn start`, tried using a symlink instead but that didn't work out of the box...
- standardizing the use of h1/h2/h3 (in markdown aligning the number of `#`)
- ran `yarn upgrade`
- hid link out icons in navbar dropdown
- added links to top navbar dropdown parent

# before hiding link icons
Screenshot 2024-05-01 at 1 30 42 PM

### CODE REVIEW

File: docs/docs/configuration/configuring-superset.mdx [hunk 1]
```diff
@@ -139,10 +161,17 @@ Use the following database drivers and connection strings:
 | [PostgreSQL](https://www.postgresql.org/) | `pip install psycopg2`    | `postgresql://<UserName>:<DBPassword>@<Database Host>/<Database Name>` |
 | [MySQL](https://www.mysql.com/)           | `pip install mysqlclient` | `mysql://<UserName>:<DBPassword>@<Database Host>/<Database Name>`      |
 
+:::tip
+Properly setting up metadata store is beyond the scope of this documentation. We recommend
+using a hosted managed service such as [Amazon RDS](https://aws.amazon.com/rds/) or
+[Google Cloud Databases](https://cloud.google.com/products/databases?hl=en) to handle
+service and supporting infrastructure and backup strategy.
+:::
+
```
Comment (rusackas) [inline docs/docs/configuration/configuring-superset.mdx:170]: I'm not sure we need all the linebreaks... is this your IDE adding these?

File: docs/docs/installation/docker-compose.mdx [hunk 2]
```diff
@@ -5,23 +5,29 @@ sidebar_position: 3
...
+import useBaseUrl from "@docusaurus/useBaseUrl";
 
-**DO NOT USE THIS FOR PRODUCTION!**
+# Using Docker Compose
 
-The fastest way to try Superset locally is using Docker Compose on a Linux or Mac OSX
-computer. Superset does not have official support for Windows. It's also the easiest
-way to launch a fully functioning **development environment** quickly.
+
+<img src={useBaseUrl("/img/docker-compose.webp" )} width="150" />
```
Comment (rusackas) [inline docs/docs/installation/docker-compose.mdx:13]: You don't see webp every day! This won't work with IE11, but that won't keep me up at night.

File: docs/docs/installation/docker-compose.mdx [hunk 3]
```diff
@@ -5,23 +5,29 @@ sidebar_position: 3
...
-and can't credibly support **high availability** as a result, we do not support nor recommend
+and can't support requirements for **high availability**, we do not support nor recommend
 using our `docker-compose` constructs to support production-type use-cases. For single host
 environments, we recommend using [minikube](https://minikube.sigs.k8s.io/docs/start/) along
 our [installing on k8s](https://superset.apache.org/docs/installation/running-on-kubernetes)
 documentation.
 :::
 
+
+As mentioned in our [quickstart guidee](/docs/quickstart), The fastest way to try
```
Comment (rusackas) [inline docs/docs/installation/docker-compose.mdx:26]:
```suggestion
As mentioned in our [quickstart guide](/docs/quickstart), The fastest way to try
```
Typo.

File: docs/docs/installation/docker-compose.mdx [hunk 4]
```diff
@@ -5,23 +5,29 @@ sidebar_position: 3
...
+Superset locally is using Docker Compose on a Linux or Mac OSX
+computer. Superset does not have official support for Windows. It's also the easiest
```
Comment (rusackas) [inline docs/docs/installation/docker-compose.mdx:28]:
```suggestion
computer. **Superset does not have official support for Windows**. It's also the easiest
```
I want to use the `<blink>` tag here.

File: docs/src/pages/index.tsx [hunk 5]
```diff
@@ -28,6 +29,8 @@ import SectionHeader from '../components/SectionHeader';
 import BlurredSection from '../components/BlurredSection';
 import '../styles/main.less';
 
+// @ts-ignore
```
Comment (rusackas) [inline docs/src/pages/index.tsx:32]: ehhhh, how bad is it? Should we fix, or sweep it under the rug?

**Review by rusackas — COMMENTED** [review]
A few tweaks/questions/comments, but generally looking good.

File: docs/docs/installation/docker-compose.mdx (hunk 2 above)
Comment (mistercrunch) [reply docs/docs/installation/docker-compose.mdx:13]: take that IE users! WHAT ARE YOU DOING ON IE?

File: docs/src/pages/index.tsx (hunk 5 above)
Comment (mistercrunch) [reply docs/src/pages/index.tsx:32]: I was getting type errors on GitHubButton not being a JSX node and was like "shut up!" Personally for the docs Im not too worried on tsc complain, either it renders or it don't...

Comment (mistercrunch) [reply docs/src/pages/index.tsx:32]: For now will sweep as i really don't feel like spending another 15 minutes looking into this and clearly that's a false negative.

File: docs/docs/configuration/configuring-superset.mdx [hunk 6]
```diff
@@ -5,28 +5,40 @@ sidebar_position: 1
...
 ---
 
-## Configuring Superset
+# Configuring Superset
 
-### Configuration
+## superset_config.py
 
-To configure your application, you need to create a file `superset_config.py`. Add this file to your
+Superset exposes hundreds of configurable parameters through its
```
[the remark this answers is not in the document]
Comment (rusackas) [reply docs/docs/configuration/configuring-superset.mdx:12]: If you think that's worth consideration, I can add it to the SIP (cc @villebro)

File: docs/.gitignore [hunk 7]
```diff
@@ -20,3 +20,6 @@ yarn-debug.log*
 yarn-error.log*
 
 docs/.zshrc
+
+# Gets copies from the root of the project at build time (yarn start / yarn build)
```
Comment (rusackas) [inline docs/.gitignore:24]:
```suggestion
# Gets copied from the root of the project at build time (yarn start / yarn build)
```

**Review by rusackas — APPROVED** [review]
Ship it (aside from maybe that one typo ;)

**Comment by mistercrunch** [comment]
ooops I merged before I saw the typo. Well PR is still net-positive typo-wize!

**Pull request merged**